### PR TITLE
OPENEUROPA-1370: Replace the functional tests with Behat in OE Media.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

In the OE Media component we have some functional javascript tests. These should instead be Behat tests agains the demo module to illustrate the provided capabilities in a way in which business can also understand what the component can offer.
h2. To do:

- [x]  Replace the 4 Functional JS tests with Behat tests like the ones that already exist
- [x]  Replace the "And I remove the media" cleanup tests with an automatic way of doing it:
 * Before and after scenario with a tag: preserveMedia
 * Before scenario that has the tag will keep track of the last Media entity ID
 * After scenario that has the tag deletes all the media entities with IDs higher than the one that was last

- [x] In the media-avportal-entity-browser feature, remove the empty lines at the end of the file (except the last one) -> this is a minor cleanup

### CODE REVIEW

File: tests/Behat/CleanupContext.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  /**
+   * Returns the IDs of all the entities of a certain type.
+   *
+   * @param string $entity_type
+   *   The entity type.
+   *
+   * @return int[]
+   *   An array of entity IDs.
+   */
+  protected function getAllEntityIdOfType(string $entity_type): array {
```
Comment (upchuk) [inline tests/Behat/CleanupContext.php]: Should be `getAllEntityIdsOfType()` - plural

File: tests/features/image.feature [hunk 2]
```diff
@@ -0,0 +1,66 @@
+@api
+Feature: Image media entities.
+  In order to show images on the website
+  As a site editor
+  I want to be able to upload images.
+
+  @cleanup:node @cleanup:media
+  Scenario: Documents can be uploaded and attached to nodes.
```
Comment (upchuk) [inline tests/features/image.feature]: Images, not documents.

File: tests/features/image.feature [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    When I go to "node/add/oe_media_demo"
+    Then I should see the heading "Create OpenEuropa Media Demo"
+    When I fill in "Title" with "My Node"
+    And I fill in the image reference field with "My Image 1"
+    And I press "Save"
+    Then I should see the heading "My Node"
+    And I should see the image "example_1.jpeg"
+
+  @javascript @cleanup:node @cleanup:media
+  Scenario: Documents can be added and referenced through the entity browser modal.
```
Comment (upchuk) [inline tests/features/image.feature]: Images, not documents.

File: tests/features/image.feature [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    Given I am logged in as a user with the "create oe_media_demo content, create image media, access media_entity_browser entity browser pages" permissions
+    And I go to "node/add/oe_media_demo"
+    Then I should see the heading "Create OpenEuropa Media Demo"
+    Given I fill in "Title" with "OpenEuropa at SymfonyCon Lisbon"
```
Comment (upchuk) [inline tests/features/image.feature]: Let's add a space line before the Given (except the first instance)

File: tests/features/image.feature [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    And I fill in "Alternative text" with "Symfonycon Lisbon"
+    And I press "Save entity"
+    And I wait for AJAX to finish
+    Then I should see the text "OpenEuropa team members at Symfonycon Lisbon"
+    And I should see the button "Remove"
+    When I press "Save"
+    Then I should see the heading "OpenEuropa at SymfonyCon Lisbon"
+    And I should see the image "example_1.jpeg"
+
+    # Reuse the existing image media into another node.
```
Comment (upchuk) [inline tests/features/image.feature]: ` Reuse the existing image media in another node.`

File: tests/features/document.feature [hunk 6]
```diff
@@ -0,0 +1,23 @@
+@api
```
Comment (upchuk) [inline tests/features/document.feature]: No entity browser tests for documents? Also, why do we have a media-file.feature and document.feature. For images we put everything in one test (also with the remote video). Let's keep some consistency and organise the tests.

File: tests/features/media-entity-browser.feature [hunk 7]
```diff
@@ -0,0 +1,16 @@
+@api
+Feature: Media entity browser.
+  In order to select the existing medias
```
Comment (upchuk) [inline tests/features/media-entity-browser.feature]: media entities not medias

File: tests/features/media-entity-browser.feature [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+  @javascript
+  Scenario: The node adding form should contain entity browser widget with possibility to view and filter all existing medias.
+    Given I am logged in as a user with the "create oe_media_demo content,access media_entity_browser entity browser pages" permission
+    And I visit "node/add/oe_media_demo"
+    And I fill in "Title" with "Media demo"
+    And I click the fieldset "Media browser field"
+    When I press the "Select entities" button
+    Then I should see entity browser modal window
+    When I click "View"
+    Then I should see search existing medias by entity browser filter
```
Comment (upchuk) [inline tests/features/media-entity-browser.feature]: This step sounds weird. `search existing medias` should be between "" (even if not used as an actual variable). Also, not medias but media entities.

File: tests/Behat/CleanupContext.php (hunk 1 above)
Comment (sergepavle) [reply tests/Behat/CleanupContext.php]: corrected

File: tests/features/image.feature (hunk 3 above)
Comment (sergepavle) [reply tests/features/image.feature]: done

File: tests/features/media-entity-browser.feature (hunk 7 above)
Comment (sergepavle) [reply tests/features/media-entity-browser.feature]: removed

File: tests/features/image.feature [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    When I fill in "Title" with "My Node"
+    And I fill in the image reference field with "My Image 1"
+    And I press "Save"
+    Then I should see the heading "My Node"
+    And I should see the image "example_1.jpeg"
+
+  @javascript @cleanup:node @cleanup:media
+  Scenario: Images can be added and referenced through the entity browser modal window.
+    Given I am logged in as a user with the "create oe_media_demo content, create image media, access media_entity_browser entity browser pages" permissions
+    And I go to "node/add/oe_media_demo"
```
Comment (nagyad) [inline tests/features/image.feature]: When

File: tests/Behat/DrupalContext.php [hunk 10]
```diff
@@ -65,4 +69,64 @@ public function removeMediaEntity(string $title): void {
...
+   * Checks that a given image is present in the page.
+   *
+   * @param string $filename
+   *   The image filename.
+   *
+   * @Then I (should )see the image :filename
+   */
+  public function assertImagePresent(string $filename): void {
+    // Drupal appends an underscore and a number to the filename when duplicate
+    // files are uploaded, for example when a test is run more than once.
```
Comment (nagyad) [inline tests/Behat/DrupalContext.php]: ...test runs more then once.

File: tests/Behat/DrupalContext.php [hunk 11]
```diff
@@ -65,4 +69,64 @@ public function removeMediaEntity(string $title): void {
...
+    // We split up the filename and extension and match for both.
+    $parts = pathinfo($filename);
+    $extension = $parts['extension'];
+    $filename = $parts['filename'];
+    $this->assertSession()->elementExists('css', "img[src$='.$extension'][src*='$filename']");
+  }
+
+  /**
+   * Checks that a OEmbed iframe for a certain url is present in the page.
```
Comment (nagyad) [inline tests/Behat/DrupalContext.php]: Checks that an OEmbed is present in the page for a certain url.

Comment (sergepavle) [reply tests/Behat/DrupalContext.php]: done

File: tests/Behat/DrupalContext.php (hunk 10 above)
Comment (sergepavle) [reply tests/Behat/DrupalContext.php]: done

File: tests/features/image.feature (hunk 9 above)
Comment (sergepavle) [reply tests/features/image.feature]: done

**Pull request merged**